fix issues with puppet6

fschaer opened the pull request:
- wrong stdlib puppet types for IP::Address::Nosubnet
- size() function now rejects undef which is the default when no slave is configured

fschaer committed:
fix issues with puppet6 :
- wrong stdlib puppet types for IP::Address::Nosubnet
- size() function now rejects undef which is the default when no slave is configured

bodgit commented:
The IP types aren't wrong here, they're not using the ones from `puppetlabs/stdlib` but from the `thrnio/ip` module where they were originally defined. If I'm going to change/update this there are more places that also need to be updated; `metadata.json` needs to be updated, (including bumping the maximum supported version of Puppet), plus any place the `thrnio/ip` module is referenced in the tests.

Also, the alignment of the class parameters has been messed up with this change.

fschaer commented:
Hi,

I completely overlooked the thrnio/ip module… strange that librarian did not pull this one here, sorry for that.

Anyway, I’ll see if I can find and replace its references with stdlib ones, if you agree, and will fix the alignement in that case (and the metadata)

bodgit commented:
They've been copied/merged into `puppetlabs/stdlib` so I should in theory update, the minimum version requirement of stdlib should be increased to the version that included them.

> strange that librarian did not pull this one here

I'm leaning more and more towards dropping librarian, it doesn't seem to be maintained much these days.

fschaer committed:
fix class params alignment

fschaer committed:
remove the thrnio/ip dependency and update metadata.json, bumping stdlib, concat and puppet min/max versions

no idea if a newer bodgitlib module would work though, so not bumping that one.

fschaer commented:
speaking of librarian... looks like the travis build failed to install deps with it, but I don't get why nor what...

fschaer commented:
Looks like not bumping bodgitlib requires stdlib < 5...

bodgit commented:
I think this is now fixed, the `thrnio/ip` module has been replaced and I've bumped the various dependency version bounds so it supports the latest stdlib, etc.
